src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -287,11 +287,21 @@ MetadataCreateCommands(void)
 		DistTableCacheEntry *cacheEntry =
 			(DistTableCacheEntry *) lfirst(distributedTableCell);
 		Oid relationId = cacheEntry->relationId;
+		Oid schemaId = InvalidOid;
+		char *createSchemaCommand = NULL;
 
 		List *workerSequenceDDLCommands = SequenceDDLCommandsForTable(relationId);
 		List *ddlCommandList = GetTableDDLEvents(relationId, includeSequenceDefaults);
 		char *tableOwnerResetCommand = TableOwnerResetCommand(relationId);
 
+		schemaId = get_rel_namespace(relationId);
+		createSchemaCommand = CreateSchemaDDLCommand(schemaId);
+		if (createSchemaCommand != NULL)
+		{
+			metadataSnapshotCommandList = lappend(metadataSnapshotCommandList,
+												  createSchemaCommand);
```

Comment:
To remain consistent, should we do an EnsureSchemaExistsOnAllNodes at the start of `start_metadata_sync_to_node` instead?

Reply:
All the callers of this function (`start_metadata_sync_to_node` and `master_metadata_snapshot`) run all the commands in a superuser connection. I did not want to open 2 sets of super user connections when these UDF's are called.

It is completely ok, if we reuse the connections opened by `EnsureSchemaExistsOnAllNodes` for the remainder of the queries for metadata syncing.

I'll check whether using EnsureSchemaExistsOnAllNodes here will increase the number of connections being opened.